### DISCUSSION TITLE
Sichtbarkeit Layer in bestimmten Maßstäben

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#102] [General]
Hallo Jürgen!

Wir haben die Anzeige eines Layers (im Beispiel Wasserspiegel) im Maßstab 1:500 - 1:20.000 konfiguriert und publiziert (siehe Screenshot der REST-Schnittstelle).

![wasserspiegel_rest](https://github.com/gis-eni/webgis-forum/assets/122986367/4c763534-0279-472e-9814-6386340050c2)

Im WebGIS wird der Layer allerdings nur bis zu einem Maßstab von ca. 1:13.000 angezeigt und danach ausgeblendet. Die Maßstabsgrenzen im CMS sind von 499 - 3.000.000 konfiguriert. 

Wasserspiegel 1:13.000

![wasserspiegel_13000](https://github.com/gis-eni/webgis-forum/assets/122986367/a3e0cc14-e7a4-4637-801e-ca6797f9aca0)

Wasserspiegel 1:13.500

![wasserspiegel_13500](https://github.com/gis-eni/webgis-forum/assets/122986367/bf311c0d-5645-40b9-bda4-d6590cbe3bab)

Die Kartenprojektion ist 3857, die Anzeige im WebGIS 31259. 
Muss bei der Konfig bzgl. Maßstabsanzeige ev. noch irgendwas beachtet werden, damit die Sichtbarkeitsgrenzen der Layer mit den Angaben im WebGIS überein stimmen? Der Layer (Wasserspiegel) soll im WebGIS bis 1:20.000 angezeigt werden und erst danach ausgeblendet werden.

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Hallo Andi,

schwer zu sagen. Normalerweise muss man da nix parametrieren. 

Hast du ein Testsystem? Was passiert, wenn du in deiner Karte einmal die Anzeige der Koordinaten auf 3857. Wenn du nicht selber testen kannst, kannst du mir den Dienst zur Verfügung stellen? Link zum Dienst per Mail bitte ;)

**Comment by gis-eni** [comment]
Das Problem ist hier folgendes:

AGS und WebGIS berechnen den Maßstab hier unterschiedlich. Wenn ich zb vom AGS einen Request(Im WebMercator) abhole, bekomme ich folgendes Ergebnis:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/9132ebc5-729b-4063-9911-dbcb3d38628f)

Der Maßstab wird aus dem Extent und der Bildbreite/Höhe berechnet:

scale = (maxx-maxy)/width * 96.0 / 0.0254  => 21163 (96dpi, 0.0254 ... von inch auf Meter). Das macht WebGIS auch, außer bei Web Mercator. Hier kommt noch die Verzerrung der durch den Längengrad ins Spiel => scale noch einmal mit COS(PHI), wobei PHI die aktuelle mittlere Breitengrad ist.

Das mit dem COS(PHI) haben wir extra (auf euren Wunsch) eingeführt, damit der Maßstab auch mit dem Maßstabsbalken zusammen passt. Ansonsten hat man zwar den Maßstabsbalken, aber wenn man am Ausdruck anfängt zu messen, passt das nicht mit dem Maßstab zusammen (wegen der Verzerrung)

Das Messen im Ausdruck ist bei WebMercator sowieso ein Problem, wie wir schon beim letzten Workshop festgestellt haben :/

Das sind halt die Probleme, die man sich mit WebMercator einfängt:

* Messen im Ausdruck eher ungenau ;)
* Unterschiedliche Systeme interpretieren den Maßstab unterschiedlich

Das zweite Problem führt dann für dieses Problem zu einer Lösung: Du musst den Dienst so anpassen, dass es in deinem Zielsystem (WebGIS) wieder passt. Sprich: den Maßstab im ArcMAP mit mit COS(PHI) dividieren (was du ja empirisch schon gemacht hast). 
Für PHI kannst du halt nur einem Mittelwert für Niederösterreich verwenden.

Alternativen:
* Maßstab in WebGIS gleich berechnen wie im AGS (dann stimmt halt der Maßstabsbalken nicht mehr, bzw stimmt schon, aber der Maßstab passt streng genommen nicht mehr dazu)
* Karten wieder in GK System anbieten

**Comment by Andreas-Lampert** [comment]
Hallo Jürgen!

Bzgl. des Problems mit der Sichtbarkeit der Layer abhängig vom Maßstab in der Projektion 3857 funktioniert es über die Anpassung der Sichtbarkeitsgrenzen im Dienst (z.B. Sichtbarkeitsgrenze im Dienst bis Maßstab 1:375.000 ergibt im WebGIS in der Projektion 3857 eine Sichtbarkeit bis zum Maßstab 1:250.000). 

Eine Fachabteilung hat jetzt allerdings angemerkt, dass die Anzeige der Legende nicht mit der Anzeige in der Karte übereinstimmt. Die Legende wird bei 1:375.000 ausgeblendet (wie im Dienst konfiguriert), die Layer in der Karte erst bei 1:250.000 (durch die Umrechnung). 

Wäre es möglich die Anzeige der Legende in der Projektion 3857 an die Anzeige der Karte anzupassen, damit Legende und Karte gleichzeitig ein- bzw. ausgeblendet werden?

Anbei noch zwei Beispiele: 1. Bild --> Gewässergüte in Karte und Legende sichtbar
2. Bild --> Gewässergüte in Karte sichtbar, in Legende nicht mehr

![legende_ok](https://github.com/gis-eni/webgis-forum/assets/122986367/60b14bfa-8f16-439a-8bae-75af2989cdbf)

![legende_nicht_ok](https://github.com/gis-eni/webgis-forum/assets/122986367/37121efa-5de1-44e7-bb50-7484cf658e66)


Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Ja, hier kommt leider zu tragen, dass alle beteiligen Komponenten intern den Maßstab unterschiedlich berechnen. Ich könnte hier im WebGIS was ändern, allerdings hätte das weitreichende Auswirkungen auf anderer Funktionen. 
Eine Umsetzung kommt daher erst einmal nicht in Frage. Für WebMercator wurde schon zu viel "herumgepfuscht", um dem Anwender einen für uns gebräuchlichen Maßstab vorzugaukeln...
Ich würde dieses Thema in die Zukunft für WebGIS 6 verschieben. Dort könnte man das WebMercator Thema noch einmal komplett neu überdenken. Vielleicht sollten wird bei dieser Projektion einfach keinen Maßstab mehr anzeigen. Ist sowieso High Risk wenn Anwender anfangen im Ausdruck auf WebMercator mit einem Prismenmaßstab herum zu messen.
